### PR TITLE
Feature/lp mint

### PR DESCRIPTION
**Pull request by carlosdimatteo**
* new mint page
* new backstop page emissions and mint banners
* optional switch Button on inputBar
* debounce hook

### CODE REVIEW

File: src/utils/comet.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      (value: string | xdr.ScVal | undefined): i128 | undefined => {
+        if (value == undefined) {
+          return undefined;
+        }
+        console.log({ value });
+        const scVal = scValToBigInt(xdr.ScVal.fromXDR(value as string, 'base64'));
+        console.log({ scVal });
+        return scVal;
+      },
+      operation
```
Comment (Ryang-21) [inline src/utils/comet.ts:51]: Going to need to convert to xdr for change that occurred in blend sdk ie operation.toXDR('base64')

File: src/pages/backstop.tsx [hunk 2]
```diff
@@ -1,34 +1,105 @@
...
+        let claimArgs: PoolClaimArgs = {
+          from: walletAddress,
+          reserve_token_ids: reserves_to_claim,
+          to: walletAddress,
+        };
+        await poolClaim(safePoolId, claimArgs, false);
+        await loadBlendData(true, safePoolId, walletAddress);
+      }
+    }
+  };
```
[the remark this answers is not in the document]
Comment (carlosdimatteo) [reply src/pages/backstop.tsx:61]: This codeblock is copied from https://github.com/blend-capital/blend-ui/blob/feature/LP-mint/src/components/dashboard/PositionOverview.tsx 
which is the claim emission button

File: src/pages/backstop.tsx [hunk 3]
```diff
@@ -78,18 +149,61 @@ const Backstop: NextPage = () => {
           ></StackedText>
         </Section>
       </Row>
-      {/*
+
+      {!!userPoolData?.estimates?.totalEmissions && (
```
[the remark this answers is not in the document]
Comment (Ryang-21) [reply src/pages/backstop.tsx:155]: The backstop user estimates object from the blend-sdk contains an emissions attribute that can be checked for whether the user has emissions to be claimed and how much they have to claim

File: src/pages/backstop.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ryang-21) [reply src/pages/backstop.tsx:61]: The code for handling submission will be similar to this. We need to replace reserves token ids with the pool addresses that we are claiming. The pool addresses that a user is accumulating emissions for is stored in the backstop user object. The wallet.tsx file contains a BackstopClaim function which will be used instead.

File: src/utils/comet.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  }
+}
+export interface cometPoolDepositArgs {
+  depositTokenAddress: string;
+  depositTokenAmount: bigint;
+  minLPTokenAmount: bigint;
+  user: string;
+}
+
+export interface competPoolGetDepositAmountByLPArgs {
```
Comment (Ryang-21) [inline src/utils/comet.ts:101]: typo

**Review by Ryang-21 — CHANGES_REQUESTED** [review]
Looks pretty much there. Found a typo and then we should also load the user's Blend balance in the userslice.tsx

**Review by Ryang-21 — APPROVED** [review]
lgtm

**Pull request merged**